scripts/update_proxy.js

```diff
@@ -0,0 +1,68 @@
+const { ethers } = require("hardhat");
+
+const PROXY_ADMIN_ABI = [{
+    "inputs": [
+        {
+            "internalType": "contract TransparentUpgradeableProxy",
+            "name": "proxy",
+            "type": "address"
+        },
+        {
+            "internalType": "address",
+            "name": "implementation",
+            "type": "address"
+        }
+    ],
+    "name": "upgrade",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+},
+{
+    "inputs": [
+        {
+            "internalType": "contract TransparentUpgradeableProxy",
+            "name": "proxy",
+            "type": "address"
+        }
+    ],
+    "name": "getProxyImplementation",
+    "outputs": [
+        {
+            "internalType": "address",
+            "name": "",
+            "type": "address"
+        }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+}];
+
+async function main() {
+    // Replace these with your actual addresses
+    const proxyContractAddress = "0x99817F89A707620591c176a1e389da70E5b9399D";
+    const proxyAdminContractAddress = "0x03A63f457420E5F58E2b86D53E68a5b2FFaC09C5";
+    const implementationAddress = '0x1c3a794a711598796b11e611a27b6E50c68B8e23';
+
+    // Need to sign using hardware wallet.
+    // const MAINNET_URL = "https://api.avax.network/ext/bc/C/rpc";
+    // const provider = new ethers.providers.JsonRpcProvider(MAINNET_URL);
+    // const signer = new LedgerSigner(provider);
+
+
+    // Create a contract instance for the proxy
+    const proxyAdmin = await ethers.getContractAt(PROXY_ADMIN_ABI, proxyAdminContractAddress, proxyAdminOwnerSigner);
```

Comment:
Hey @isa16alejandra, I've updated this update proxy according to your comment. I have two doubts:

1. Where this `proxyAdminOwnerSigner` comes from?
2. I would need to sign this with my Ledger hardware wallet, since it is the one registered as the proxy owner. Does it make sense to you the lines 49 and 50?

Reply:
1. **proxyAdminOwnerSigner** should be the owner of the proxyAdmin contract. So you can replace it with the signer of the Ledger.

2. Not sure if this is the correct way to integrate with the ledger (never done it before), I was checking this [documentation](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-ledger), they said you need to change the hardhat.config.js. In any case what you can do is before running the script on avalanche you do it on a forked network. Let me know if you need some help

Reply:
> proxyAdminOwnerSigner should be the owner of the proxyAdmin contract. So you can replace it with the signer of the Ledger.

I'm still a little lost about this one, this is not an address, right?